src/Plugin/tmgmt/Translator/ThebigwordTranslator.php

```diff
@@ -0,0 +1,872 @@
+    $this->setTranslator($job->getTranslator());
+
+    try {
+      $job_id = $job->id();
+      $project_id = $this->newTranslationProject($job_id, $job_id, $job->getSetting('required_by'), $job->getSetting('quote_required'), $job->getSetting('category'), $job->getSetting('review'));
+      $job->addMessage('Created a new project in thebigword with the id: @id', ['@id' => $project_id], 'debug');
+
+      foreach ($job->getItems() as $job_item) {
+        /** @var RemoteMapping $remote_mapping */
+        $remote_mapping = \Drupal::entityTypeManager()
+          ->getStorage('tmgmt_remote')
+          ->create([
+            'tjid' => $job->id(),
+            'tjiid' => $job_item->id(),
+            'data_item_key' => 'tmgmt_thebigword',
+            'remote_identifier_1' => 'ProjectId',
+            'remote_identifier_2' => $project_id,
+            'files' => [],
+          ]);
+        $remote_mapping->save();
+        $this->sendFiles($job_item);
+      }
+      // Confirm is required to trigger the translation.
+      $confirmed = $this->confirmUpload($project_id, 'ReferenceAdd');
+      if ($confirmed != count($job->getItems())) {
+        $message = 'Not all the references had been confirmed.';
+        throw new TMGMTException($message);
+      }
+      $confirmed = $this->confirmUpload($project_id, 'TranslatableSource');
+      if ($confirmed != count($job->getItems())) {
+        $message = 'Not all the sources had been confirmed.';
+        throw new TMGMTException($message);
+      }
+
+      $job->submitted('Job has been successfully submitted for translation.');
+      $job->settings->project_id = $project_id;
+    }
+    catch (TMGMTException $e) {
+      $this->sendFileError('RestartPoint03', '', $e->getMessage(), $job, NULL, TRUE);
+      \Drupal::logger('tmgmt_thebigword')->error('Job has been rejected with following error: @error',
+        array('@error' => $e->getMessage()));
+      $job->rejected('Job has been rejected with following error: @error',
+        array('@error' => $e->getMessage()), 'error');
+    }
+  }
+
+  /**
+   * Does a request to Thebigword services.
+   *
+   * @param string $path
+   *   Resource path.
+   * @param string $method
+   *   HTTP method (GET, POST...).
+   * @param array $params
+   *   Form parameters to send to Thebigword service.
+   * @param bool $download
+   *   If we expect resource to be downloaded.
+   *
+   * @return array
+   *   Response array from Thebigword.
+   *
+   * @throws \Drupal\tmgmt\TMGMTException
+   */
+  public function request($path, $method = 'GET', $params = array(), $download = FALSE) {
+    $options = array();
+    if (!$this->translator) {
+      throw new TMGMTException('There is no Translation entity. Access to public/secret keys is not possible.');
+    }
+
+    if (\Drupal::config('tmgmt_thebigword.settings')->get('use_mock_service')) {
+      $url = $GLOBALS['base_url'] . '/tmgmt_thebigword_mock' . '/v' . self::API_VERSION . '/' . $path;
```

Comment:
this would then go away, instead, just read the URL from config and in a test, we'd set the config.